### PR TITLE
[v6][Feature] View components for menus

### PR DESCRIPTION
**Pull request by tabacitu**
### BEFORE - What was wrong? What was happening before this PR?

If an admin panel had multiple themes installed and switched between them, they had to switch menu blade files too. Each theme had to have its own menu, because the HTML syntax for creating a menu item, menu dropdown, menu separator is slightly different between each Bootstrap theme.

More info in https://github.com/Laravel-Backpack/CRUD/issues/5134 which this PR fixes.

### AFTER - What is happening after this PR?

Developers can do this within their `menu_items.blde.php`:

```php
<x-backpack::menu-item title="Dashboard" icon="la la-dashboard" :link="backpack_url('dashboard')" />

<x-backpack::menu-separator title="First-Party Addons" />

<x-backpack::menu-dropdown title="News" icon="la la-newspaper-o">
    <x-backpack::menu-dropdown-item title="Articles" icon="la la-newspaper-o" :link="backpack_url('article')" />
    <x-backpack::menu-dropdown-item title="Categories" icon="la la-list" :link="backpack_url('category')" />
    <x-backpack::menu-dropdown-item title="Tags" icon="la la-tag" :link="backpack_url('tag')" />
</x-backpack::menu-dropdown>
```

### How did you achieve that, in technical terms?

Added 4 [view components](https://laravel.com/docs/10.x/blade#components):
- `MenuItem`
- `MenuDropdown`
- `MenuDropdownItem`
- `MenuSeparator`

These components render their content using `backpack_view('path_to_blade_file')` which means they look into all Backpack theme namespaces: `view_namespace` then `fallback_view_namespace` then `backpack.ui` (ultimate fallback).

That means:
- Backpack CRUD provides the default views in `resources/views/ui/components`
- themes can override those `ui` view by providing a few blade files in their `resources/views/components` directory;
- themes can even override the PHP classes if they want, to provide more functionality (by binding to them); but I don't really see a reason why they would at the moment;

### Is it a breaking change?

No, it's non-breaking.

### How can we test the before & after?

This doesn't need testing at the moment. Just a little more thinking - whether this is a good idea or not. Whether we should continue development or not. Right now here's a before-and-after of a menu:

CleanShot 2023-06-21 at 12 21 32

I think it's pretty similar, which I like. And you can easily mix-and-match these components and normal HTML, which I like.

But `x-backpack::menu-separator ` is pretty ugly, that I don't like. But there's no good way around it.

### TODOs

- [x] @pxpm & @tabacitu to think about possible issues this might bring up (no ballooning);
- [x] @tabacitu implement a `visibleWhen()` method on components (most difficult part here is naming);
- [x] @tabacitu create PRs with component views for all 3 themes;
- [ ] @pxpm test and review (without ballooning the scope);

### CODE REVIEW

**Comment by tabacitu** [comment]
I just did the 3 PRs for the themes too 🤦‍♂️ But now they're all pretty:
- https://github.com/Laravel-Backpack/theme-tabler/pull/79
- https://github.com/Laravel-Backpack/theme-coreuiv4/pull/16
- https://github.com/Laravel-Backpack/theme-coreuiv2/pull/9

To test this, please:


**Step 1.** Use the branch `view-components-for-menus` on:
- CRUD
- theme-tabler
- theme-coreuiv2
- theme-coreuiv4

**Step 2.** Delete all the sidebar_content files in `reources/views/vendor/backpack/theme-xx/inc` (all 3 themes)

**Step 3.** Copy-paste this to your `resources/views/vendor/backpack/ui/inc/menu_items.blade.php`:

```php
{{-- This file is used for menu items by any Backpack v6 theme --}}

<x-backpack::menu-item title="Dashboard" icon="la la-dashboard" :link="backpack_url('dashboard')" />

@includeWhen(class_exists(\Backpack\DevTools\DevToolsServiceProvider::class), 'backpack.devtools::buttons.sidebar_item')

<x-backpack::menu-separator title="First-Party Addons" />

<x-backpack::menu-dropdown title="News" icon="la la-newspaper-o">
    <x-backpack::menu-dropdown-item title="Articles" icon="la la-newspaper-o" :link="backpack_url('article')" />
    <x-backpack::menu-dropdown-item title="Categories" icon="la la-list" :link="backpack_url('category')" />
    <x-backpack::menu-dropdown-item title="Tags" icon="la la-tag" :link="backpack_url('tag')" />
</x-backpack::menu-dropdown>

<!-- Users, Roles Permissions -->
<x-backpack::menu-dropdown title="Authentication" icon="la la-group">
    <x-backpack::menu-dropdown-item title="Users" icon="la la-user" :link="backpack_url('user')" />
    <x-backpack::menu-dropdown-item title="Roles" icon="la la-group" :link="backpack_url('role')" />
    <x-backpack::menu-dropdown-item title="Permissions" icon="la la-key" :link="backpack_url('permission')" />
</x-backpack::menu-dropdown>

<!-- Advanced -->
<x-backpack::menu-dropdown title="Other" icon="la la-cogs">
    <x-backpack::menu-dropdown-item title="File Manager" icon="la la-files-o" :link="backpack_url('elfinder')" />
    <x-backpack::menu-dropdown-item title="Backups" icon="la la-hdd-o" :link="backpack_url('backup')" />
    <x-backpack::menu-dropdown-item title="Logs" icon="la la-terminal" :link="backpack_url('log')" />
    <x-backpack::menu-dropdown-item title="Settings" icon="la la-cog" :link="backpack_url('setting')" />
    <x-backpack::menu-dropdown-item title="Pages" icon="la la-file-o" :link="backpack_url('page')" />
    <x-backpack::menu-dropdown-item title="Menu" icon="la la-list" :link="backpack_url('menu-item')" />
</x-backpack::menu-dropdown>

<x-backpack::menu-separator title="Example CRUDs" />

<!-- Monsters -->
<x-backpack::menu-dropdown title="Monsters & Stuff" icon="la la-optin-monster">
    <x-backpack::menu-dropdown-item title="Monsters" icon="la la-optin-monster" :link="backpack_url('monster')" />
    <x-backpack::menu-dropdown-item title="Caves" icon="la la-dungeon" :link="backpack_url('cave')" />
    <x-backpack::menu-dropdown-item title="Stories" icon="la la-book" :link="backpack_url('story')" />
    <x-backpack::menu-dropdown-item title="Icons" icon="la la-info-circle" :link="backpack_url('icon')" />
    <x-backpack::menu-dropdown-item title="Products" icon="la la-shopping-cart" :link="backpack_url('product')" />
    <x-backpack::menu-dropdown-item title="Fluent Monsters" icon="la la-pastafarianism"
        :link="backpack_url('fluent-monster')" />
    <x-backpack::menu-dropdown-item title="Field Monsters" icon="la la-list-alt"
        :link="backpack_url('field-monster')" />
    <x-backpack::menu-dropdown-item title="Editable Monsters" icon="la la-spell-check"
        :link="backpack_url('editable-monster')" />
    <x-backpack::menu-dropdown-item title="Dummies" icon="la la-poo" :link="backpack_url('dummy')" />
</x-backpack::menu-dropdown>

<!-- Pets -->
<x-backpack::menu-dropdown title="Pet Shop" icon="la la-dog">
    <x-backpack::menu-dropdown-item title="About" icon="la la-question" :link="backpack_url('pet-shop/about')" />
    <x-backpack::menu-dropdown-item title="Invoices" icon="la la-file-text" :link="backpack_url('pet-shop/invoice')" />
    <x-backpack::menu-dropdown-item title="Owners" icon="la la-user" :link="backpack_url('pet-shop/owner')" />
    <x-backpack::menu-dropdown-item title="Pets" icon="la la-dog" :link="backpack_url('pet-shop/pet')" />
    <x-backpack::menu-dropdown-item title="Badges" icon="la la-certificate" :link="backpack_url('pet-shop/badge')" />
    <x-backpack::menu-dropdown-item title="Skills" icon="la la-ribbon" :link="backpack_url('pet-shop/skill')" />
    <x-backpack::menu-dropdown-item title="Comments" icon="la la-comment" :link="backpack_url('pet-shop/comment')" />
</x-backpack::menu-dropdown>

```

**Comment by pxpm** [comment]
Hey @tabacitu I think there is one big thing missing here, **the ability to pass arbitrary attributes like (rel, target etc)**. 

I think this is super important for sidebar/menu (not always a sidebar) items.

**Comment by tabacitu** [comment]
I've just implemented the `:visible` attribute in https://github.com/Laravel-Backpack/CRUD/pull/5146 - proving it can be done in a non-breaking way. But I'm not gonna merge it just yet - it's unimportant, and we have no time for a difficult decision with no real value added right now. Let's move on with the launch 💪

Going to merge this.

**Comment by tabacitu** [comment]
Added docs. Changed generators to use them. We're done here 🎉

File: src/app/View/Components/MenuDropdown.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    /**
+     * Create a new component instance.
+     */
+    public function __construct(
+        public ?string $title = null,
+        public ?string $icon = null,
+        public ?string $link = null,
+        public array $items = [],
+    ) {
+    }
```
Comment (promatik) [inline src/app/View/Components/MenuDropdown.php:20]: 🧽✨

**Pull request merged**